pci/src/vfio.rs

```diff
@@ -376,6 +376,19 @@ impl VfioCommon {
             let region_size: u64;
             let bar_addr: GuestAddress;
 
+            let base_addr = if let Some(resources) = &resources {
+                if resources.len() > ranges.len() {
+                    return Err(PciDeviceError::MissingResource);
+                }
```

Comment:
> IIUC, this should be `if resources.len() < ranges.len()`, right?

This is trying to check that there is not going to be a bounds failure on the slice access below. A nicer way to to it would be to use `match resources.get(ranges.len()).ok_or(PciDeviceError::MissingResource)? { ...`

Reply:
Let me have a try with this approach.